apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/sdl/GraphSDLSchemaParser.kt

```diff
@@ -17,38 +18,207 @@ import java.io.IOException
+  deprecationReason: String
+}
+
+""${'"'}
+Arguments provided to Fields or Directives and the input fields of an InputObject are represented as Input Values which describe their type and optionally a default value.
+""${'"'}
+type __InputValue {
+  name: String!
+  description: String
+  type: __Type!
+
+  ""${'"'}
+  A GraphQL-formatted string representing the default value for this input value.
+  ""${'"'}
+  defaultValue: String
+}
+
+""${'"'}
+A GraphQL Schema defines the capabilities of a GraphQL server. It exposes all available types and directives on the server, as well as the entry points for query, mutation, and subscription operations.
+""${'"'}
+type __Schema {
+  ""${'"'}A list of all types supported by this server.""${'"'}
+  types: [__Type!]!
+
+  ""${'"'}The type that query operations will be rooted at.""${'"'}
+  queryType: __Type!
+
+  ""${'"'}
+  If this server supports mutation, the type that mutation operations will be rooted at.
+  ""${'"'}
+  mutationType: __Type
+  ""${'"'}
+  If this server supports subscription, the type that subscription operations will be rooted at.
+  ""${'"'}
+  subscriptionType: __Type
+  ""${'"'}A list of all directives supported by this server.""${'"'}
+  directives: [__Directive!]!
+}
+""${'"'}
+The fundamental unit of any GraphQL Schema is the type. There are many kinds of types in GraphQL as represented by the `__TypeKind` enum.
+
+Depending on the kind of a type, certain fields describe information about that type. Scalar types provide no information beyond a name and description, while Enum types provide their values. Object and Interface types provide the fields they describe. Abstract types, Union and Interface, provide the Object types possible at runtime. List and NonNull types compose other types.
+""${'"'}
+type __Type {
+  kind: __TypeKind!
+  name: String
+  description: String
+  fields(includeDeprecated: Boolean = false): [__Field!]
+  interfaces: [__Type!]
+  possibleTypes: [__Type!]
+  enumValues(includeDeprecated: Boolean = false): [__EnumValue!]
+  inputFields: [__InputValue!]
+  ofType: __Type
+}
+""${'"'}An enum describing what kind of type a given `__Type` is.""${'"'}
+enum __TypeKind {
+  ""${'"'}Indicates this type is a scalar.""${'"'}
+  SCALAR
+  ""${'"'}
+  Indicates this type is an object. `fields` and `interfaces` are valid fields.
+  ""${'"'}
+  OBJECT
+  ""${'"'}
+  Indicates this type is an interface. `fields` and `possibleTypes` are valid fields.
+  ""${'"'}
+  INTERFACE
+  ""${'"'}Indicates this type is a union. `possibleTypes` is a valid field.""${'"'}
+  UNION
+  ""${'"'}Indicates this type is an enum. `enumValues` is a valid field.""${'"'}
+  ENUM
+  ""${'"'}
+  Indicates this type is an input object. `inputFields` is a valid field.
+  ""${'"'}
+  INPUT_OBJECT
+  ""${'"'}Indicates this type is a list. `ofType` is a valid field.""${'"'}
+  LIST
+  ""${'"'}Indicates this type is a non-null. `ofType` is a valid field.""${'"'}
+  NON_NULL
+}
+  """.trimIndent()
```

Comment:
Adding the built-in introspection types was required for the `introspection_query` codegen test to pass. More generally, users should be able to generate models for introspection types if needed.

I started adding these types programmatically but it becomes verbose real fast and we have a parser for that already so I put the builtin types as inline SDL

Reply:
Ha, very smart :) Nice move. Do you think if it worth to move this to separate file?

Reply:
Yes, I can make it a resource, this should work as long as we keep `apollo-compiler` JVM